app/lib/features/link_room/pages/link_room_page.dart

```diff
@@ -0,0 +1,205 @@
+import 'package:acter/common/extensions/options.dart';
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/providers/space_providers.dart';
+import 'package:acter/common/widgets/search.dart';
+import 'package:acter/common/widgets/sliver_scaffold.dart';
+import 'package:acter/features/home/widgets/space_chip.dart';
+import 'package:acter/features/link_room/types.dart';
+import 'package:acter/features/link_room/widgets/link_space_list_item.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::space::link_room_sheet');
+
+class LinkRoomPage extends ConsumerStatefulWidget {
+  static const confirmJoinRuleUpdateKey = Key('link-room-confirm-join-rule');
+  static const denyJoinRuleUpdateKey = Key('link-room-deny-join-rule');
+
+  final String parentSpaceId;
+  final ChildRoomType childRoomType;
+
+  const LinkRoomPage({
+    super.key,
+    required this.parentSpaceId,
+    required this.childRoomType,
+  });
+
+  @override
+  ConsumerState<LinkRoomPage> createState() => _LinkRoomPageConsumerState();
+}
+
+class _LinkRoomPageConsumerState extends ConsumerState<LinkRoomPage> {
+  final TextEditingController searchTextEditingController =
+      TextEditingController();
+
```

Comment:
removed a bunch of stateful ness here that could be done with providers instead. Later realized that we can even collapse most of these providers.